### ISSUE TITLE
New Rule: Prefer `LOWER(column) LIKE '%match%'` to `column ILIKE '%Match%'`

### ISSUE THREAD

**Issue by jpy-git** [sqlfluff/sqlfluff#2396]
From GitLab SQL style guide: https://about.gitlab.com/handbook/business-technology/data-team/platform/sql-style-guide/#general

Prefer `LOWER(column) LIKE '%match%'` to `column ILIKE '%Match%'`. This lowers the chance of stray capital letters leading to an unexpected result

Additionally there are quite a few posts about it being faster performance wise:
- https://til.hashrocket.com/posts/e65bb49204-lower-is-faster-than-ilike
- https://stackoverflow.com/questions/20336665/lower-like-vs-ilike
- https://dba.stackexchange.com/questions/89901/lower-vs-ilike-on-postgresql

**Comment by sti0** [comment]
Quick note: `ILIKE` is not supported in `Exasol` dialect.

**Comment by jpy-git** [comment]
@sti0 thanks! This rule would switch from ILIKE to LOWER so this shouldn't have any risk of corrupting exasol

**Comment by sti0** [comment]
Yes. Just a information for you for your journey learning the Exasol dialect 😄
